### ISSUE TITLE
bug(COMPONENT): Accessibility error: the <th> for the divider in month view and year views (in table header) contains no text.

### ISSUE THREAD

**Issue by jumbo6** [angular/components#28509]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

WAVE tool reports the accessibility error (WCAG A) as "Empty table header" to datepicker calendar popup views (month, year, multi-year). 

An empty <th> for the divider causes the error.
![image](https://github.com/angular/components/assets/116878/543804f4-010c-42d5-ba4d-573abacf4b47)

### Reproduction

Steps to reproduce:

To replicate the issue, 

1. go to Datepicker examples page: https://material.angular.io/components/datepicker/overview

2. click a calendar icon to popup the calendar view. Then open WAVE extension.
![image](https://github.com/angular/components/assets/116878/228adc65-cfd3-49e4-8b9b-a57d772f3fbd)

![image](https://github.com/angular/components/assets/116878/9155bbf8-b920-4c27-8d04-179343125479)

### Expected Behavior

in <th>, use a hidden text to describe the divider. 

### Actual Behavior

To replicate the issue, go to Datepicker examples page, click a calendar icon to popup the calendar view. Then open WAVE extension.

https://material.angular.io/components/datepicker/overview

### Environment

- Angular:
- CDK/Material:
- Browser(s):
- Operating System (e.g. Windows, macOS, Ubuntu):

**Comment by zarend** [comment]
Hello @jumbo6 ,

Thank you for bringing this to our attention. I can see how a table header could be inappropriate if a table is only used for layout. It could be inappropriate to communicate to screen readers that there's a header if the `th` is not intended to actually communicate a header.

Sometimes AXE, WAVE and other a11y checkers or linters give false positives.

In the Datepicker month-view, the calendar is an ARIA grid role and the header element is aria-hidden. By looking at the Chrome a11y tree, we can see that the header is not included in the a11y tree, which means that assistive technology does not see it. 

<img width="1691" alt="image" src="https://github.com/angular/components/assets/7720245/74c4cb18-25f6-4de7-a68c-aa9d8ed234ca">

It may be that the WAVE checker always reports when there is an empty th element, even if it does not actually communicate semantics to screen readers.

I do not think that this check is causing an actual accessibility problem for users, and I believe it is safe to ignore it.

Have you been seeing reports from users of your applications that this is causing a problem?

Best Regards,

Zach

**Comment by jumbo6** [comment]
RE: I do not think that this check is causing an actual accessibility problem for users, and I believe it is safe to ignore it.

Thank you for your response. I haven't heard anything from the end users. This is the report from the testers. 

Other than the empty header, there are 2 empty labels in mat-calendar-body.html... 


<label [id]="_startDateLabelId" class="mat-calendar-body-hidden-label">
  {{startDateAccessibleName}}
</label>
<label [id]="_endDateLabelId" class="mat-calendar-body-hidden-label">
  {{endDateAccessibleName}}
</label>

can you add a condition like ngif="startDateAccessibleName", so the label is removed when there is no use of it?

**Comment by zarend** [comment]
Hello folks,

Thanks for getting back to us. Angular components doesn't have a requirement to cleanly pass automated accessibility checks.

This isn't something I'm able to take on right now, but we have this issue report to track this.

Best Regards,

Zach

**Comment by zarend** [comment]
#27241

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
